mbed_devices/_internal/windows/component_descriptor.py

```diff
@@ -0,0 +1,123 @@
+"""Defines a generic Win32 component."""
+import logging
+from abc import ABC, abstractmethod
+from typing import List, Any, Generator, Optional
+
+import win32com.client
+
+UNKNOWN_VALUE = "Unknown"
+NAMED_TUPLE_FIELDS_ATTRIBUTE = "_fields"
+
+logger = logging.getLogger(__name__)
+
+
+class ComponentDescriptor(ABC):
```

Comment:
I'm really struggling to understand what this object does and why we need all of this complexity. It seems like some sort of abstract factory class which instantiates a `NamedTuple` in a bunch of different ways, with some unexpected uses of `@classmethod`. Do we really need to make this so complex? I'd prefer some simple factory functions over this abstract factory (if that's what this is).

Reply:
I radically changed the code to make it simpler to follow. Hopefully, I succeeded.